### ISSUE TITLE
redundant_guards could lint against more checks in the standard library

### ISSUE THREAD

**Issue by sgued** [rust-lang/rust-clippy#11807]
### Summary

The [redundant_guards](https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_guards) lint could lint against more checks than it does.

Currently it only lints against `if <capture> == pattern`, but it could also lint against some `if` guards that use `std` functions, notably `str::is_empty` and `slice::is_empty`.

### Lint Name

redundant_guards

### Reproducer

I tried this code:

```rust
fn inner() -> Option<(&'static str> {
    todo!();
}
let a = inner();
match a {
    Some(s) if s.is_empty() => todo!("1"),
    Some(_) => todo!("5"),
    None => todo!("4")
}
```

I expected the `redundant_guard` lint to trigger and suggest using `Some("") => todo!("1"),` in the first `match` arm.

Instead, the lint didn't trigger.

### Version

```text
Nightly version: 1.76.0-nightly

(2023-11-13 ba7c7a301984967c8c13)

(From playground)
```

**Comment by y21** [comment]
Similarily:
- `slice.is_empty()` can be `[]`
- `slice.starts_with(<pat_like>)` can be `[<pat_like>, ..]`
- `slice.ends_with(<pat_like>)` can be `[.., <pat_like>]`

@rustbot claim
